### PR TITLE
Feature/issue 937 flatten include meta.hpp

### PR DESCRIPTION
**Pull request by drezap**
## Summary
Replaces all meta includes with one meta.hpp include.

## Tests
Same tests, all pass.

## Checklist

- [x] Math issue #937

- [x] Copyright holder: 2019 Andre Zapico

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

## Remaining TODO

- [x] remove the 5 shell scripts
- [x] remove direct meta include from stan/math/prim/arr/functor/coupled_ode_observer.hpp
- [x] Add the meta include and remove direct includes from stan/math/prim/mat/fun/cov_matrix_free_lkj.hpp.

### CODE REVIEW

File: stan/math/prim/arr/functor/coupled_ode_observer.hpp [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
 #define STAN_MATH_PRIM_ARR_FUNCTOR_COUPLED_ODE_OBSERVER_HPP
 
+#include <stan/math/prim/meta.hpp>
 #include <stan/math/prim/scal/err/check_size_match.hpp>
 #include <stan/math/prim/scal/err/check_less.hpp>
-#include <stan/math/prim/scal/meta/is_constant_struct.hpp>
-#include <stan/math/prim/scal/meta/operands_and_partials.hpp>
-#include <stan/math/prim/scal/meta/broadcast_array.hpp>
 #include <stan/math/prim/arr/fun/sum.hpp>
 #include <stan/math/prim/scal/meta/return_type.hpp>
```
Comment (syclik) [inline stan/math/prim/arr/functor/coupled_ode_observer.hpp]: This file still has a meta include: `#include <stan/math/prim/scal/meta/return_type.hpp>`.  This should be removed.

**Comment by syclik** [comment]
Awesome!!! Yeah, so this will be good once:
1. remove the 5 shell scripts
2. remove direct meta include from `stan/math/prim/arr/functor/coupled_ode_observer.hpp`
3. Add the meta include and remove direct includes from stan/math/prim/mat/fun/cov_matrix_free_lkj.hpp.

I'll update the top-level comment with a checklist.

**Comment by drezap** [comment]
alright we're all good, thanks for going through that.

**Comment by drezap** [comment]
Good catch, I've changed the `stack_alloc.hpp`, the `fwd/core/fvar.hpp` won't pass tests if I include any combination of `fwd/meta.hpp` `prim/meta.hpp`

**Comment by drezap** [comment]
This has been approved by @roualdes. @syclik, if you don't see anything wrong with this, I don't see why we shouldn't merge.

**Comment by syclik** [comment]
Yup. Go ahead and merge!

**Comment by drezap** [comment]
I tried to merge from my phone and github took away my merge privileges. Can someone else merge this?

**Comment by rok-cesnovar** [comment]
Done.

**Comment by drezap** [comment]
ty!

**Comment by bob-carpenter** [comment]
> On Jun 19, 2019, at 2:36 PM, Andre Zapico <notifications@github.com> wrote:
> 
> I tried to merge from my phone and github took away my merge privileges. Can someone else merge this?

Are you sure you were logged in?  I'm constantly having to re-login on my phone and iPad.

**Comment by drezap** [comment]
> Are you sure you were logged in?

Not at all

**Comment by seantalts** [comment]
Github doesn't let you merge your own pull requests from your phone as far as I know. I have been annoyed by this in the past :joy:

**Pull request merged**